### PR TITLE
Fix gcp login initially loading error no roles incorrectly

### PR DESCRIPTION
**Pull request by jimmytsang**
Problem: gcp sso page was not processing the resourceAccessList fetch correctly thus initially showing "Error you don't have any gcp roles..."

Solution: dispatch loading, loading success, and loading failure in the redux `getResourceAccessList` and render the `ReduxLoader`

### CODE REVIEW

File: ui/src/pages/gcp/login.js [hunk 1]
```diff
@@ -303,62 +298,61 @@ class GCPLoginPage extends React.Component {
...
+                    </p>
+                    <p>
+                        If that does not work, please ask your Athenz domain
+                        admin for assistance.
+                    </p>
+                </ParentWrapperDiv>
+            </div>
+        );
+        return this.props.isLoading.length !== 0 ? (
+            <ReduxPageLoader message={'Loading microsegmentation data'} />
```
Comment (abvaidya) [inline ui/src/pages/gcp/login.js:352]: This error message doesnt look correct. Shouldnt it be something like "Loading GCP projects"?

**Review by abvaidya — CHANGES_REQUESTED** [review]

File: ui/src/pages/gcp/login.js [hunk 2]
```diff
@@ -303,62 +298,61 @@ class GCPLoginPage extends React.Component {
...
+                    <h3>
+                        Error: There are no GCP project roles associated with
+                        your account.
+                    </h3>
+                    <p>
+                        Check to make sure that your <tt>gcp.*</tt> roles
+                        contain your user.
+                        <br />
+                        E.g., if your account is 'jdoe', then{' '}
+                        <tt>gcp.fed.admin.user</tt> should have{' '}
```
Comment (havetisyan) [inline ui/src/pages/gcp/login.js:341]: this is a yahoo specific role so we should not included it here. Let's keep it generic and just refer to that your account should be in the configured federated roles that have access to the project and not list any specific role names.

Comment (jimmytsang) [reply ui/src/pages/gcp/login.js:341]: how's this? 
image
```
Error: There are no GCP project roles associated with your account.
Check to make sure that your gcp.* roles contain your user. Your account should be in the configured federated roles that have access to the project. If that does not work, please ask your Athenz domain admin for assistance.
```

File: ui/src/pages/gcp/login.js (hunk 1 above)
Comment (jimmytsang) [reply ui/src/pages/gcp/login.js]: great catch, fixed.

**Review by abvaidya — APPROVED** [review]

**Pull request merged**